rotkehlchen/rotkehlchen.py

```diff
@@ -555,23 +556,82 @@ def get_history_query_status(self) -> Dict[str, str]:
 
     def process_history(
             self,
+            report_id: int,
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> Tuple[Dict[str, Any], str]:
-        (
-            error_or_empty,
-            history,
-            loan_history,
-            asset_movements,
-            eth_transactions,
-            defi_events,
-            ledger_actions,
-        ) = self.events_historian.get_history(
-            start_ts=start_ts,
-            end_ts=end_ts,
-            has_premium=self.premium is not None,
-        )
+        def process_cache(cache_data: List[Dict[str, Any]]) -> Union[None, HistoryResult]:
```

Comment:
The return type seems wrong here. The function always returns HistoryResult. And don't do `Union[None, XXX]`. It should be `Optional[XXX]`